core/integration/container_test.go

```diff
@@ -3982,3 +3983,22 @@ func TestContainerImageLoadCompatibility(t *testing.T) {
 		}
 	}
 }
+
+func TestContainerWithPlainSecretVariable(t *testing.T) {
+	t.Parallel()
+
+	c, ctx := connect(t)
+	defer c.Close()
+
+	ctr := c.Container().From("alpine").
+		WithPlainSecretVariable("SECRET_ENV", "secret-value")
+
+	// Assert env value cannot be read from API
+	v, err := ctr.EnvVariable(ctx, "SECRET_ENV")
+	require.NoError(t, err)
+	require.Equal(t, "", v)
```

Comment:
Good assertion 👍

Does it really return an empty string when env not found? We should probably return an error there. Same with `Label` and other calls like it. (needs an issue)

Reply:
In a shell, if you type `echo $VAR_THAT_DOES_NOT_EXIST`, nothing will be print so maybe we should keep it that way?
Or we could have an option `required`? In some CI it might happened that depending on the presence or not of an environment variable, something happens differently.

What do you think?

Reply:
But you're using `echo` so makes sense that it doesn't error. But if you do `printenv NOPE` you do get an exit code 1.

The problem is that env variables with an empty value are totally valid, but the API already returns the right thing: `null` (https://play.dagger.cloud/playground/i4w90IwPozQ). Python returns `None` but not Go because codegen is declaring optional strings as `string`. It should be `*string` so that it can return `nil` when not set.

Reply:
I see! I'll pop up on issue

Reply:
Issue created #5449

Reply:
I wasn't clear. I was trying to say that the API already does the right thing, as well as Python. Meaning that it returns `null` instead of an error or an empty string.